pkg/virt-handler/virtiofs/dispatcher_suite_test.go

```diff
@@ -0,0 +1,11 @@
+package virtiofs
```

Comment:
```suggestion
package virtiofs_test
```

Should be in package `virtiofs_test`

Reply:
If I change this, I'll not be able to unit test the dispatcher, since I can replace getPid and execCommand without exporting them

Reply:
btw, I know this is not the "right thing" but the alternative is just to remove the unit test (I'll not expose changing those function in the API) and just rely on the functional one, wdyt?

Reply:
@germag we can use an interface for those function. Something like:
```golang
//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE
type manager interface {
	ExecCommand(string, ...string) *exec.Cmd
	GetPid(string) (int, error)
}

type VirtiofsManager struct {
	mountBaseDir string
        manager manager
}

func (m *VirtiofsManager) ExecCommand(...) *exec.Cmd {
	return exec.Command(....)
}

func (m *VirtiofsManager) GetPid(socket string) (int, error) {
	return isolation.GetPid(socket string)
}
```

You can find a concrete example in https://github.com/kubevirt/kubevirt/pull/14309/commits/8ffa163e9a90ead79b15a45d5b2bbcf2efd55192#diff-058be21453873277e69df581ea1d141ff82b7fb3b10af1b7c5a0ca5c4bebcf2e

Reply:
Yes, but I'm trying to avoid exposing in the module api the possibility to change those functions, because outside testing it has no use, so is just confusing. 

Also, IIUC it means the mock now is part of the module's exported api so it will be compiled in the release, even if it is only used for testing, while anything that is defined inside a `_test.go` file is ignored.

Reply:
If the main idea is just to test the public API, I prefer to drop the unit test I rely only on the functional test

Reply:
>Also, IIUC it means the mock now is part of the module's exported api so it will be compiled in the release, even if it is only used for testing, while anything that is defined inside a _test.go file is ignored.

Not sure if the mocks are compiled into the release. @0xFelix do you know?

Reply:
>> If the main idea is just to test the public API, I prefer to drop the unit test I rely only on the functional test
> No you are testing how StartVirtiofsDispatcher behaves based on what getpid and exec return. I wouldn't drop the unit tests

I did a bad job explaining myself here, sorry.

What I meant was, if requiring the test to be in their own package, like `virtiofs_test`, is to "force" tests to only account for the public API, that's fine. But, extending the public API just to accommodate the test (in this case by allowing to replace those function) is defying that purpose, making the api more confusing. So, I think is more simple and clear to keep the test in the same package, and go will not build them because of the `_test.go` suffix.

Reply:
> Not sure if the mocks are compiled into the release. @0xFelix do you know?

I don't know if it can be done differently or if there is another place to define which file should not be part of the build, but IIUC your example(*), `generated_mock_multipath_monitor.go` will be compiled, AFAIK go only ignores `*_test.go` files, so that code will be included in the build

(*) https://github.com/kubevirt/kubevirt/commit/8ffa163e9a90ead79b15a45d5b2bbcf2efd55192#diff-c2d45a9b4e705989225e20cb3ba474b0f568b66216911bd00a82c710063a7f25

Reply:
also in the BUILD.bazel:
```
go_library(
    name = "go_default_library",
    srcs = [
        "generated_mock_multipath_monitor.go",
        "multipath_monitor.go",
    ],

```

so the mock is part of the build

Reply:
Depending on the `-package` flag passed to mockgen, the mocks can be placed in a different package. However, I'm not sure this works with the version of mockgen KubeVirt is using at the moment.

Reply:
The package is not what makes it into the build or not, it's the file name, more specifically the file name suffix

Reply:
> I expect that it should be possible to modify that by changing the value that is passed to the -destination flag.

that is what I said here: https://github.com/kubevirt/kubevirt/pull/14115#discussion_r2028877620

But, the mocking is not the issue here, so I don0t want to derail the conversation (although just skimming through the mock uses in kubevirt, seem to have the same problem)